### PR TITLE
feat: keep filter order

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Screenshot 2023-12-19 at 13 51 20

* usePerisistentTableState state maintains URL query params order and also maintains the same order when storing object in localStorage
* Filters component respects the order of keys in the state object that it gets

## Discussion points

Reviewer: please test those changes locally with local storage and url refresh

### CODE REVIEW

File: frontend/src/component/feature/FeatureToggleList/FeatureToggleListTable.tsx [hunk 1]
```diff
@@ -95,14 +95,6 @@ const FeatureToggleListTableComponent: VFC = () => {
         stateConfig,
     );
 
-    const filterState = {
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureToggleList/FeatureToggleListTable.tsx:98]: this object was breaking state ordering

File: frontend/src/component/filter/Filters/Filters.tsx [hunk 2]
```diff
@@ -90,9 +90,16 @@ export const Filters: VFC<IFilterProps> = ({
     };
 
     useEffect(() => {
-        const newSelectedFilters = availableFilters
+        const newSelectedFilters = Object.keys(state)
```
Comment (kwasniew) [inline frontend/src/component/filter/Filters/Filters.tsx:93]: work off the state, not the available filters

File: frontend/src/hooks/usePersistentTableState.test.tsx [hunk 3]
```diff
@@ -229,4 +232,29 @@ describe('usePersistentTableState', () => {
...
+                queryParamsDefinition={{
+                    query: StringParam,
+                    another: StringParam,
+                    ignore: StringParam,
+                }}
+            />,
+            { route: '/my-url?another=another&query=initialUrl' },
+        );
+
+        expect(screen.getByTestId('state-keys').textContent).toBe(
```
Comment (kwasniew) [inline frontend/src/hooks/usePersistentTableState.test.tsx:251]: respect url not query params definition as before

File: frontend/src/utils/reorderObject.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+export const reorderObject = <T extends object>(obj: T, order: string[]): T => {
```
Comment (kwasniew) [inline frontend/src/utils/reorderObject.ts:1]: utility to reorder state object with query params order

**Review by sjaanus — APPROVED** [review]

**Pull request merged**